### PR TITLE
Snap 2878

### PR DESCRIPTION
**Pull request by smahajan05**
## Changes proposed in this pull request

Added test cases for sql functions which are covered in this document:

https://docs.google.com/document/d/1tYBSKYSvFJM0kfi4Q5Z2sgg1irN5V-J_zE6LJbiIxxI/edit?usp=sharing

Some of the sql function tests are failing with plan caching on. Some test failures are not related to plan caching. Added exception handling for all those functions which are throwing exception.

## Patch testing

Verified the changes by running SQLFunctionsTestSuite.scala

## ReleaseNotes.txt changes

NA

## Other PRs 

None

### CODE REVIEW

**Comment by PradeepSurale** [comment]
Can one of the admins verify this patch?

File: cluster/src/test/scala/org/apache/spark/sql/SQLFunctionsTestSuite.scala [hunk 1]
```diff
@@ -0,0 +1,2773 @@
...
+      pw: PrintWriter, sqlContext: SparkSession): Any = {
+
+    var snappyDF = snc.sql(snappyQuery)
+    val snappyQueryFileName = s"Snappy_$queryNum.out"
+    val sparkQueryFileName = s"Spark_$queryNum.out"
+    val snappyDest = getTempDir("snappyQueryFiles_" + tableType, onlyOnce = false)
+    val sparkDest = getTempDir("sparkQueryFiles", onlyOnce = true)
+    val sparkFile = new File(sparkDest, sparkQueryFileName)
+    val snappyFile = new File(snappyDest, snappyQueryFileName)
+    val col1 = snappyDF.schema.fieldNames(0)
```
Comment (hemanthmeka) [inline cluster/src/test/scala/org/apache/spark/sql/SQLFunctionsTestSuite.scala]: Are col1, col, cols used anywhere?

File: cluster/src/test/scala/org/apache/spark/sql/SQLFunctionsTestSuite.scala [hunk 2]
```diff
@@ -0,0 +1,2773 @@
...
+    }
+  }
+
+  def assertQueryFullResultSet(snc: SnappyContext, sparkQuery: String,
+      snappyQuery: String, numRows: Int,
+      queryNum: String, tableType: String,
+      pw: PrintWriter, sqlContext: SparkSession): Any = {
+
+    var snappyDF = snc.sql(snappyQuery)
+    val snappyQueryFileName = s"Snappy_$queryNum.out"
```
Comment (hemanthmeka) [inline cluster/src/test/scala/org/apache/spark/sql/SQLFunctionsTestSuite.scala]: Can we compare spark and snappy results without writing to files and reading them back?

File: cluster/src/test/scala/org/apache/spark/sql/SQLFunctionsTestSuite.scala (hunk 1 above)
Comment (smahajan05) [reply cluster/src/test/scala/org/apache/spark/sql/SQLFunctionsTestSuite.scala]: Removed unused variables

File: cluster/src/test/scala/org/apache/spark/sql/SQLFunctionsTestSuite.scala (hunk 2 above)
Comment (smahajan05) [reply cluster/src/test/scala/org/apache/spark/sql/SQLFunctionsTestSuite.scala]: Added different approach for comparing spark and snappy results. Added code for comparing snappy and spark df.

**Pull request merged**